http-gateway/pkg/server/gateway_integration_test.go

```diff
@@ -143,18 +142,11 @@ func post(port int, path string, body io.Reader, headerKV ...string) *http.Respo
 	return resp
 }
 
-func waitForHttpGatewayToBeReady() {
-	attempts := 20
-	durationBetweenAttempts := time.Millisecond * 100
+func waitForHttpGatewayToBeReady(port int) {
+	url := fmt.Sprintf("http://localhost:%d", port)
 
-	for i := 0; i < attempts; i++ {
-		_, err := net.Dial("tcp", ":http")
-		if err != nil {
-			fmt.Print(".")
-			time.Sleep(durationBetweenAttempts)
-			continue
-		}
-
-		break
-	}
+	Eventually(func() error {
+		_, err := http.Get(url)
```

Comment:
There is no handler for the path `/`, so I'm not convinced this will work or, if it does work some of the time, whether it will always work. I was thinking it might be better to wrap the POST requests in the test in `Eventually`.

Reply:
I can step down to the previous tactic of dialling the socket instead of performing a full HTTP `GET` request. Not sure I want to add logic to fix a test.

Reply:
See my comment about hitting the health handler